app/controllers/cleaners_controller.rb

```diff
@@ -37,6 +44,27 @@ def create
     end
   end
 
+  def confirm_token
+    @cleaner = Cleaner.find_by(confirm_token: params[:token])
+    if @cleaner
+      @cleaner.update!(email_confirmed: true, confirm_token: nil)
+      respond_to do |format|
+        format.html { redirect_to 'confirm_token', notice: "Confirmation Completed" }
+      end
+    end
+  end
+
+  def confirmed
+    # empty
+  end
+
+  def add_cities(cleaner_id)
+    city_ids = params[:cleaner][:city_ids]
+    city_ids.each do |city|
+      CleanersCity.create!(city_id: city, cleaner_id: cleaner_id) if !(city_ids.nil?)
```

Comment:
No "city_ids.present?" is not working, instead of it,  i use..

unless city.blank?